experimental/traffic-portal/src/app/api/cache-group.service.ts

```diff
@@ -428,13 +429,34 @@ export class CacheGroupService extends APIService {
 	}
 
 	/**
-	 * Deletes an existing asn.
+	 * Replaces the current definition of a ASN with the one given.
 	 *
-	 * @param asnOrId Id of the asn to delete.
-	 * @returns The deleted asn.
+	 * @param asn The new ASN.
+	 * @returns The updated ASN.
 	 */
-	public async deleteASN(asnOrId: number | ResponseASN): Promise<void> {
-		const id = typeof(asnOrId) === "number" ? asnOrId : asnOrId.id;
+	public async updateASN(asn: ResponseASN): Promise<ResponseASN> {
+		const path = `asns/${asn.id}`;
+		return this.put<ResponseASN>(path, asn).toPromise();
+	}
+
+	/**
+	 * Creates a new ASN.
+	 *
+	 * @param asn The ASN to create.
+	 * @returns The created ASN.
+	 */
+	public async createASN(asn: RequestASN): Promise<ResponseASN> {
+		return this.post<ResponseASN>("asns", asn).toPromise();
+	}
+
+	/**
+	 * Deletes an existing ASN.
+	 *
+	 * @param asnID ID of the ASN to delete.
```

Comment:
This argument is either the ASN itself, or the ID of the ASN to be deleted, not just its ID.

Reply:
it is the ID of the ASN to be deleted.. I am not sure what would you like me to change?